### PR TITLE
 make nodeJarUrl configurable for Google Compute Engine nodesource

### PR DESCRIPTION
**Pull request by tao-xinxiu**

### CODE REVIEW

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java [hunk 1]
```diff
@@ -40,16 +39,25 @@
...
 
-    private static Configuration nsConfig = null;
+    private static Configuration nsConfig;
+
+    static {
+        try {
+            // If the configuration manager is not loaded, I load it with the NodeSource properties file
+            nsConfig = NSProperties.loadConfig();
+        } catch (ConfigurationException e) {
+            // If something go wring, I switch to hardcoded configuration, and leave.
```
Comment (MaelAudren) [inline infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java]: wring => wrong

File: infrastructures/infrastructure-gce/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/GCEInfrastructure.java [hunk 2]
```diff
@@ -200,49 +200,49 @@ private void validate(Object[] parameters) {
...
         if (parameters[parameterIndex] == null) {
-            parameters[parameterIndex] = DEFAULT_REGION;
+            throw new IllegalArgumentException("The connector-iaas URL must be specified");
         }
-        // ram
+        // nodeJarUrl
         parameterIndex++;
         if (parameters[parameterIndex] == null) {
-            parameters[parameterIndex] = String.valueOf(DEFAULT_RAM);
+            throw new IllegalArgumentException("The url for downloading the node jar must be specified");
```
Comment (MaelAudren) [inline infrastructures/infrastructure-gce/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/GCEInfrastructure.java]: Be consistent in your abbreviations. In the upper exception, you use `URL` and here `url`. I think using `URL` is better in this case.

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java [hunk 3]
```diff
@@ -108,14 +115,7 @@ private String generateNodeStartCommand(String instanceId, String rmUrlToUse, St
 
     public String generateDefaultIaasConnectorURL(String DefaultRMHostname) {
         if (nsConfig == null) {
```
Comment (nebilbenmabrouk) [inline infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java]: I suggest to remove the whole if block as it is useless. I prefer the mode fail fast when nsConfig is null

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java [hunk 4]
```diff
@@ -124,26 +124,15 @@ public String generateDefaultIaasConnectorURL(String DefaultRMHostname) {
 
     public String generateDefaultDownloadCommand(String rmHostname) {
         if (nsConfig == null) {
```
Comment (nebilbenmabrouk) [inline infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java]: Same here, I suggest to remove the if block.

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java [hunk 5]
```diff
@@ -124,26 +124,15 @@ public String generateDefaultIaasConnectorURL(String DefaultRMHostname) {
...
 
-    private static void loadNSConfig() {
-        try {
-            if (null == nsConfig) {
-                nsConfig = NSProperties.loadConfig();
-            }
-        } catch (ConfigurationException e) {
-            logger.error("Exception when loading NodeSource properties", e);
+    public String generateDefaultNodeJarURL(String rmHostname) {
+        if (nsConfig == null) {
```
Comment (nebilbenmabrouk) [inline infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java]: Same here, I suggest to remove the if block.

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java (hunk 3 above)
Comment (tao-xinxiu) [reply infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java]: Done, I've added `throw new RuntimeException(e);` when catch `ConfigurationException` during loadConfig.

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java (hunk 1 above)
Comment (tao-xinxiu) [reply infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/LinuxInitScriptGenerator.java]: Done and adapted the comment.

File: infrastructures/infrastructure-gce/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/GCEInfrastructure.java (hunk 2 above)
Comment (tao-xinxiu) [reply infrastructures/infrastructure-gce/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/GCEInfrastructure.java]: Done.

**Pull request merged**